internal/repo/user.go

```diff
@@ -0,0 +1,61 @@
+package repo
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dexguitar/chatapp/internal/model"
+	pg "github.com/snaffi/pg-helper"
+)
+
+type UserRepository struct{}
+
+func NewUserRepo() *UserRepository {
+	return &UserRepository{}
+}
+
+func (u *UserRepository) CreateUser(ctx context.Context, user *model.User, db pg.DB) error {
+	op := "UserRepository.CreateUser"
+
+	q := "INSERT INTO users (username, email, password) VALUES ($1, $2, $3)"
+
+	_, err := db.Exec(ctx, q, user.Username, user.Email, user.Password)
+	if err != nil {
+		return fmt.Errorf("%s: %w", op, err)
+	}
+
+	return nil
+}
+
+func (u *UserRepository) FindUserByCreds(ctx context.Context, username, password string, db pg.Read) (*model.User, error) {
+	op := "UserRepository.FindUserByCreds"
+
+	var user model.User
+
+	q := "SELECT * FROM users WHERE username = $1 AND password = $2"
```

Comment:
you should enumerate columns explicitly because order of columns is not specified.